realm/realm-library/src/androidTest/java/io/realm/BulkInsertTests.java

```diff
@@ -954,4 +954,29 @@ public void insertOrUpdate_collection_notInTransaction() {
     public void insertOrUpdate_object_notInTransaction() {
         realm.insert(new AllTypes());
     }
+
+
+    @Test
+    public void insertOrUpdate_fromOtherRealm() {
```

Comment:
Not sure I understand this test? It doesn't seem to be related to the primary key change?